Goagen panics (nil pointer) with parent resource

## Expected Behavior
```
$ goagen bootstrap -d github.com/coincircle/api/design
app
app/contexts.go
app/controllers.go
app/hrefs.go
app/media_types.go
app/user_types.go
app/test
app/test/block_explorerv1_testing.go
app/test/transactionsv1_testing.go
app/test/usersv1_testing.go
app/test/walletsv1_testing.go
app/test/health_testing.go
tool/cli
tool/cli/commands.go
client
client/client.go
client/block_explorerv1.go
client/transactionsv1.go
client/usersv1.go
client/walletsv1.go
client/health.go
client/user_types.go
client/media_types.go
swagger
swagger/swagger.json
swagger/swagger.yaml
```

## Actual Behavior
```
$ goagen bootstrap -d github.com/coincircle/api/design
exit status 2
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x6f9c5d]

goroutine 1 [running]:
github.com/goadesign/goa/design.(*dupper).DupAttribute(...)
        /home/adam/go/src/github.com/goadesign/goa/design/dup.go:40
github.com/goadesign/goa/design.(*dupper).DupType(0xc4202514e0, 0xaa77c0, 0xc420286840, 0xc420251550, 0xc4202514f0)
        /home/adam/go/src/github.com/goadesign/goa/design/dup.go:67 +0x1bd
github.com/goadesign/goa/design.Dup(0xaa77c0, 0xc420286840, 0xc420251820, 0xc420251900)
        /home/adam/go/src/github.com/goadesign/goa/design/dup.go:7 +0x154
github.com/goadesign/goa/design.(*ActionDefinition).initQueryParams(0xc420210960)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:1662 +0x2bd
github.com/goadesign/goa/design.(*ActionDefinition).Finalize(0xc420210960)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:1532 +0x98
github.com/goadesign/goa/design.(*ResourceDefinition).Finalize.func2(0xc420210960, 0xc4202865d0, 0x85778a)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:891 +0x38
github.com/goadesign/goa/design.(*ResourceDefinition).IterateActions(0xc42022a380, 0xc420251c38, 0x0, 0x0)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:722 +0x1e2
github.com/goadesign/goa/design.(*ResourceDefinition).Finalize(0xc42022a380)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:885 +0x126
github.com/goadesign/goa/dslengine.finalizeSet(0xc4201fd040, 0x5, 0x5, 0xc4201fd040, 0x5)
        /home/adam/go/src/github.com/goadesign/goa/dslengine/runner.go:330 +0x8e
github.com/goadesign/goa/design.(*APIDefinition).IterateSets(0xc4201b0b00, 0x88e338)
        /home/adam/go/src/github.com/goadesign/goa/design/definitions.go:506 +0x6ae
github.com/goadesign/goa/dslengine.Run(0xa9f580, 0xafae20)
        /home/adam/go/src/github.com/goadesign/goa/dslengine/runner.go:114 +0x244
main.main()
        /home/adam/go/src/github.com/coincircle/api/goagen548223936/main.go:24 +0x5a

Makefile:5: recipe for target 'goa' failed
make: *** [goa] Error 1
```
## Steps to Reproduce the Problem (including complete and simple design if relevant)
You can find my design file, below. I can get bootstrap to successfully generate my API if I comment out the two `Parent(...)` functions.

```
package design

import (
	. "github.com/goadesign/goa/design"        // dot imports are standard for GOA
	. "github.com/goadesign/goa/design/apidsl" // dot imports are standard for GOA
)

var _ = API("coincircle", func() {
	Title("Coincircle API")

	Description("API to interact with all of Coincircle's services")

	Host("api.coincircle.com")

	Scheme("https")

	Consumes("application/json")
	Produces("application/json")

	Origin("*", func() {
		Methods("GET", "POST")
	})
})

/********************
 *
 * RESOURCES
 *
 */

var _ = Resource("health", func() {
	BasePath("/health")

	Action("ping", func() {
		Description("Check API status")
		Routing(GET("ping"))
		Response(OK)
	})
})

var _ = Resource("BlockExplorerV1", func() {
	BasePath("/block-explorer/v1")

	Response(BadRequest, StandardErrorMedia)
	Response(InternalServerError, StandardErrorMedia)
	Response(NotFound, StandardErrorMedia)

	Action("blocks", func() {
		Routing(GET("blocks"))

		Description("Query blocks")

		Params(func() {
			Param("from", Integer, "Min block number")
			Param("to", Integer, "Max block number")
		})

		Response(OK, ArrayOf(BlockMedia))
		Response(BadRequest, StandardErrorMedia)
		Response(Gone, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
	})
	Action("block", func() {
		Routing(GET("blocks/:blockNumberOrHash"))

		Description("Get a single block by block number or hash")

		Params(func() {})

		Response(OK, BlockMedia)
		Response(BadRequest, StandardErrorMedia)
		Response(Gone, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
	})
	Action("transactions", func() {
		Routing(GET("transactions"))

		Description("Query blockchain transactions")

		Params(func() {
			Param("fromBlock", Integer, "Min block number")
			Param("toBlock", Integer, "Max block number")
			Param("input", String, "Only return items where input equals some value")
			Param("inputOperator", String, "One of [STARTSWITH, ENDSWITH, CONTAINS, EQUALS]. Defaults to 'EQUALS'")
			Param("from", String, "Only match transactions where the from address equals this value")
			Param("to", String, "Only match transactions where the to address equals this value")
		})

		Response(OK, ArrayOf(TransactionMedia))
		Response(BadRequest, StandardErrorMedia)
		Response(Gone, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
	})
	Action("transaction", func() {
		Routing(GET("transactions/:hash"))

		Description("Get a single transaction by hash")

		Params(func() {
			Param("includeLogs", Boolean, "If true, includes an array of log objects")
		})

		Response(OK, TransactionMedia)
		Response(BadRequest, StandardErrorMedia)
		Response(Gone, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
	})
	Action("logs", func() {
		Routing(GET("logs"))

		Description("Get contract event logs")

		Params(func() {
			Param("fromBlock", Integer, "Min block number")
			Param("toBlock", Integer, "Max block number")
			Param("addresses", ArrayOf(String), "Only match logs emitted by this address. Supports multiple addresses by specifying the same query param multiple times")
			Param("topics0", ArrayOf(String), "Query log topics0. Supports multiple addresses by specifying the same query param multiple times")
			Param("topics1", ArrayOf(String), "Query log topics1. Supports multiple addresses by specifying the same query param multiple times")
			Param("topics2", ArrayOf(String), "Query log topics2. Supports multiple addresses by specifying the same query param multiple times")
			Param("topics3", ArrayOf(String), "Query log topics3. Supports multiple addresses by specifying the same query param multiple times")
		})

		Response(OK, ArrayOf(LogMedia))
		Response(BadRequest, StandardErrorMedia)
		Response(Gone, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
	})
})

var _ = Resource("TransactionsV1", func() {
	Parent("WalletsV1")
	BasePath("/transactions")

	Action("list", func() {
		Description("Get all of a user's non-pending transactions")

		Routing(GET(""))

		Params(func() {
			Param("perPage", Integer, func() {
				Description("The maximum number of transactions to return per page")
				Example(20)
				Minimum(1)
				Maximum(100)
			})
			Param("page", Integer, func() {
				Description("The page number to return (one indexed)")
				Example(10)
				Minimum(1)
			})
			Param("sort", String, func() {
				Description("Sorts the transactions by timestamp (DESC = newest to oldest)")
				Enum("DESC", "ASC")
			})

			Required("perPage", "page", "sort")
		})

		Response(OK, ArrayOf(TransactionMedia))
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID or walletID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user or wallet with that id does not exist")
		})
	})

	Action("show", func() {
		Description("Get a transaction by hash")

		Routing(GET("/:txHash"))

		Response(OK, TransactionMedia)
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID or walletID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user or wallet with that id does not exist")
		})
	})

	Action("new", func() {
		Description("Create a new transaction")

		Routing(POST(""))

		Payload(NewTransactionPayload)

		Response(OK, func() {
			Media(TransactionMedia, "pending")
		})
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID or walletID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user or wallet with that id does not exist")
		})
	})

	Action("new token transaction", func() {
		Description("Create a new token transaction. Note: this is a simplified version of the 'new transaction' end point specifically for sending tokens, but either can be used to send tokens. Do not use this end point to send ethereum.")

		Routing(POST("/tokens"))

		Payload(NewTokenTransactionPayload)

		Response(OK, func() {
			Media(TransactionMedia, "pending")
		})
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID or walletID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user or wallet with that id does not exist")
		})
	})
})

var _ = Resource("UsersV1", func() {
	BasePath("/users/v1")

	Action("show", func() {
		Routing(GET("/:userID"))

		Description("Fetch a user by id")

		Response(OK, UserMedia)
		Response(BadRequest, StandardErrorMedia)
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID")
		})
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user with that id does not exist")
		})
	})

	Action("new", func() {
		Description("Create a new user")

		Routing(POST(""))

		Payload(NewUserPayload)

		Response(OK, UserMedia)
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(Gone, StandardErrorMedia, func() {
			Description("A user already exists in our system. You should follow the 'Authorization Code' grant type oAuth 2.0 flow in order to request permissions to be able to act on the user's behalf")
		})
	})
})

var _ = Resource("WalletsV1", func() {
	Parent("UsersV1")
	BasePath("/wallets")

	Action("list", func() {
		Description("List all of a user's wallets")

		Routing(GET(""))

		Params(func() {
			Param("perPage", Integer, func() {
				Description("The maximum number of transactions to return per page")
				Example(20)
				Minimum(1)
				Maximum(100)
			})
			Param("page", Integer, func() {
				Description("The page number to return (one indexed)")
				Example(10)
				Minimum(1)
			})
			Param("sort", String, func() {
				Description("Sorts the transactions by timestamp (DESC = newest to oldest)")
				Enum("DESC", "ASC")
			})
			Param("balances", ArrayOf(String), func() {
				Description("An array of token contract address for which to return available balances. For ETH, use 'ETH'.")
				Example([]string{"ETH", "0x0d8775f648430679a709e98d2b0cb6250d2887ef", "0x24692791bc444c5cd0b81e3cbcaba4b04acd1f3b"})
			})

			Required("perPage", "page", "sort")
		})

		Response(OK, ArrayOf(WalletMedia))
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user with that id does not exist")
		})
	})

	Action("show", func() {
		Description("Get a user's wallet")

		Routing(GET("/:walletID"))

		Params(func() {
			Param("balances", ArrayOf(String), func() {
				Description("An array of token contract address for which to return available balances. For ETH, use 'ETH'.")
				Example([]string{"ETH", "0x0d8775f648430679a709e98d2b0cb6250d2887ef", "0x24692791bc444c5cd0b81e3cbcaba4b04acd1f3b"})
			})
		})

		Response(OK, WalletMedia)
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID or walletID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(NotFound, StandardErrorMedia, func() {
			Description("A user or wallet with that id does not exist")
		})
	})

	Action("new", func() {
		Description("Create a new wallet")

		Routing(POST(""))

		Payload(NewWalletPayload)

		Response(OK, WalletMedia)
		Response(Unauthorized, StandardErrorMedia)
		Response(Forbidden, StandardErrorMedia, func() {
			Description("The client is not authorized to act on the provided userID")
		})
		Response(BadRequest, StandardErrorMedia)
		Response(InternalServerError, StandardErrorMedia)
		Response(Gone, StandardErrorMedia, func() {
			Description("If the maximum number of wallets have been created for this user for your clientID, you will not be able to create another")
		})
	})
})

/********************
 *
 * MEDIATYPES
 *
 */

// AddressValidationMedia is the media type returned when validating an address
var AddressValidationMedia = MediaType("application/address_validation+json", func() {
	Description("Price")

	Attributes(func() {
		Attribute("valid", Boolean, "Address is validated")

		Required("valid")
	})

	View("default", func() {
		Attribute("valid")
	})
})

// BlockMedia is the standard ethereum block type
var BlockMedia = MediaType("application/block+json", func() {
	Description("An ethereum block")

	Attributes(func() {
		Attribute("difficulty", String, "Block difficulty")
		Attribute("extraData", String, "Extra data")
		Attribute("gasLimit", String, "Gas limit")
		Attribute("gasUsed", String, "Gas used")
		Attribute("hash", String, "Block hash")
		Attribute("logsBloom", String, "Logs bloom")
		Attribute("miner", String, "Miner of block")
		Attribute("mixHash", String, "Mix hash")
		Attribute("nonce", String, "Nonce")
		Attribute("number", Integer, "Block Number")
		Attribute("parentHash", String, "Parent hash")
		Attribute("receiptsRoot", String, "Receipts root")
		Attribute("sha3Uncles", String, "SHA3 Uncles")
		Attribute("size", String, "Block size")
		Attribute("stateRoot", String, "State root")
		Attribute("timestamp", Integer, "UNIX Timestamp")
		Attribute("totalDifficulty", String, "Total difficulty")
		Attribute("transactionsRoot", String, "Transactions Root")
	})

	View("default", func() {
		Attribute("difficulty")
		Attribute("extraData")
		Attribute("gasLimit")
		Attribute("gasUsed")
		Attribute("hash")
		Attribute("logsBloom")
		Attribute("miner")
		Attribute("mixHash")
		Attribute("nonce")
		Attribute("number")
		Attribute("parentHash")
		Attribute("receiptsRoot")
		Attribute("sha3Uncles")
		Attribute("size")
		Attribute("stateRoot")
		Attribute("timestamp")
		Attribute("totalDifficulty")
		Attribute("transactionsRoot")
	})
})

// LogMedia represents an ethereum block event
var LogMedia = MediaType("application/log+json", func() {
	Description("Log")

	Attributes(func() {
		Attribute("address", String, "Contract address which emitted the event")
		Attribute("blockHash", String, "Block hash in which the event was emitted")
		Attribute("blockNumber", Integer, "Block number")
		Attribute("data", String, "Input data")
		Attribute("logIndex", Integer, "Log index")
		Attribute("removed", Boolean, "True if the transaction was removed")
		Attribute("transactionHash", String, "Transaction hash in which the event was emitted")
		Attribute("transactionIndex", Integer, "Transaction index")
		Attribute("logTopic0", String, "Log Topic 0")
		Attribute("logTopic1", String, "Log Topic 1")
		Attribute("logTopic2", String, "Log Topic 2")
		Attribute("logTopic3", String, "Log Topic 3")
	})

	View("default", func() {
		Attribute("address")
		Attribute("blockHash")
		Attribute("blockNumber")
		Attribute("data")
		Attribute("logIndex")
		Attribute("removed")
		Attribute("transactionHash")
		Attribute("transactionIndex")
		Attribute("logTopic0")
		Attribute("logTopic1")
		Attribute("logTopic2")
		Attribute("logTopic3")
	})
})

// RoleMedia defines roles, which are assigned to user to limit or grant permissions
var RoleMedia = MediaType("application/role+json", func() {
	Description("Roles are used for authorization purposes and are assigned to users (e.g. 'user' or 'admin')")

	Attributes(func() {
		Attribute("id", UUID, func() {
			Description("v4 uuid")
			Example("739ea65c-9b7a-4dbc-9515-3578314a2790")
		})

		Attribute("name", String, func() {
			Description("A human readable description")
			Enum("user", "admin")
		})

		Required("id", "name")
	})

	View("default", func() {
		Attribute("id")
		Attribute("name")
	})
})

// StandardErrorMedia defines the standard error type.
var StandardErrorMedia = MediaType("application/standard_error+json", func() {
	Description("A standard error response")

	Attributes(func() {
		Attribute("code", Integer, "A code that describes the error", func() {
			Example(400)
		})
		Attribute("message", String, "A message that describes the error", func() {
			Example("Bad Request")

		})

		Required("code", "message")
	})

	View("default", func() {
		Attribute("code")
		Attribute("message")
	})
})

// StatusCodeMedia is used to provide more information for cetain responses
var StatusCodeMedia = MediaType("application/status_code+json", func() {
	Description("Status Code")

	Attributes(func() {
		Attribute("code", String, "Code")
		Attribute("message", String, "Message")
	})

	Required("code", "message")

	View("default", func() {
		Attribute("code")
		Attribute("message")
	})
})

// TransactionMedia defines a typical ethereum transaction
var TransactionMedia = MediaType("application/transaction+json", func() {
	Description("Transaction")

	Attributes(func() {
		Attribute("timestamp", Integer, "Timestamp")
		Attribute("blockHash", String, "Block hash")
		Attribute("blockNumber", Integer, "Block Number")
		Attribute("from", String, "From address")
		Attribute("to", String, "To address")
		Attribute("gas", String, "Gas limit")
		Attribute("gasPrice", String, "Gas price")
		Attribute("hash", String, "Transaction hash")
		Attribute("input", String, "Input data")
		Attribute("nonce", String, "Nonce")
		Attribute("transactionIndex", String, "Transaction index")
		Attribute("value", String, "Value in wei")
		Attribute("v", String, "V param of ECDSA")
		Attribute("r", String, "R param of ECDSA")
		Attribute("s", String, "S param of ECDSA")
		Attribute("gasUsed", String, "Gas used")
		Attribute("contractAddress", String, "Deployed contract address (If any)")
	})

	View("default", func() {
		Attribute("timestamp")
		Attribute("blockHash")
		Attribute("blockNumber")
		Attribute("from")
		Attribute("to")
		Attribute("gas")
		Attribute("gasPrice")
		Attribute("hash")
		Attribute("input")
		Attribute("nonce")
		Attribute("transactionIndex")
		Attribute("value")
		Attribute("v")
		Attribute("r")
		Attribute("s")
		Attribute("gasUsed")
		Attribute("contractAddress")
	})

	View("pending", func() {
		Attribute("timestamp")
		Attribute("from")
		Attribute("to")
		Attribute("gas")
		Attribute("gasPrice")
		Attribute("hash")
		Attribute("input")
		Attribute("nonce")
		Attribute("value")
		Attribute("contractAddress")
	})
})

// UserMedia defines a CoinCircle user
var UserMedia = MediaType("application/user+json", func() {
	Description("Users are end-users of the CoinCircle family of products")

	Attributes(func() {
		Attribute("id", UUID, func() {
			Description("v4 uuid")
			Example("2364acf1-203b-46d7-a4ab-859a6e5e64ba")
		})
		Attribute("email", String, func() {
			Description("Email (case insensitive)")
			Example("Ursula@CoinCircle.com")
		})
		Attribute("role", RoleMedia, "Used for authorization purposes")

		Required("id", "email", "role")
	})

	View("default", func() {
		Attribute("id")
		Attribute("email")
		Attribute("role")
	})
})

// WalletMedia is the standard wallet object response
var WalletMedia = MediaType("application/wallet+json", func() {
	Description("Wallets are used by users to send and receive transactions on the blockchain")

	Attributes(func() {
		Attribute("id", UUID, func() {
			Description("v4 uuid")
			Example("1af2ba31-b12b-431d-9bdb-511002bd85aa")
		})
		Attribute("address", String, func() {
			Description("The wallet's public address")
			Example("0x1e09373fb7dcfd6475f4c76e0cdb89de3d585e55")
		})
		Attribute("balances", HashOf(String, String), func() {
			Description("An object of token balances, where keys represent the token addresses and values represent balances.")
			Example(map[string]string{
				"ETH": "0x9184e72a",
				"0x0d8775f648430679a709e98d2b0cb6250d2887ef": "0x9184e72a",
			})
		})

		Required("id", "address")
	})

	View("default", func() {
		Attribute("id")
		Attribute("address")
	})
	View("balances", func() {
		Attribute("id")
		Attribute("address")
		Attribute("balances")
	})
})

/********************
 *
 * PAYLOADS
 *
 */

// NewUserPayload is sent to the new user end point to create a user
var NewUserPayload = Type("NewUserPayload", func() {
	Attribute("email", String, func() {
		Description("The email for the user that you are registering. Emails must be unique and are case insensitive")
		Example("Ursula@CoinCircle.com")
	})

	Required("email")
})

// NewWalletPayload is sent to the new wallet end point to create a new wallet for a user
var NewWalletPayload = Type("NewWalletPayload", func() {
	Attribute("protocol", String, func() {
		Description("The protocol of the wallet to create. Note: the protocol should be 'ETH' for any tokens transacted on the ethereum blockchain.")
		Enum("ETH")
	})

	Required("protocol")
})

// NewTokenTransactionPayload is sent to the new token ransaction end point to send a transaction for a user
var NewTokenTransactionPayload = Type("NewTokenTransactionPayload", func() {
	Attribute("gas", String, func() {
		Description("The maximum gas allowed to be spent to process the transaction. If not provided, an acceptable value will be provided for you.")
		Example("0x76c0")
	})
	Attribute("gasPrice", String, func() {
		Description("The price used to determine the total gas cost. If not provided, an acceptable value will be provided for you.")
		Example("0x9184e72a000")
	})
	Attribute("nonce", String, func() {
		Description("The transaction sequence number from the sending account. If not provided, this will be calculated for you.")
		Example("0x76c0")
	})
	Attribute("recipientAddress", String, func() {
		Description("The public wallet address of the recipient")
		Example("0x765010b525D050c33104aE663317c723c90158bB")
	})
	Attribute("tokenAddress", String, func() {
		Description("The smart contract address of the token being sent")
		Example("0xd46e8dd67c5d32be8058bb8eb970870f072445675")
	})
	Attribute("tokens", String, func() {
		Description("The number of tokens to send, in the units of the token (e.g. ETH, not WEI for ethereum)")
		Example("0x9184e72a")
	})

	Required("recipientAddress", "tokenAddress", "tokens")
})

// NewTransactionPayload is sent to the new transaction end point to send a transaction for a user
var NewTransactionPayload = Type("NewTransactionPayload", func() {
	Attribute("data", String, func() {
		Description("The standard 'txdata' for an ethereum transaction (https://github.com/ethereum/go-ethereum/blob/55599ee95d4151a2502465e0afc7c47bd1acba77/core/types/transaction.go#L57). Please note that this is required for sending a token. If you'd like to send a token, but don't know how to/ want to include this, please see our dedicated 'send token' end point.")
		Example("0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675")
	})
	Attribute("gas", String, func() {
		Description("The maximum gas allowed to be spent to process the transaction. If not provided, an acceptable value will be provided for you.")
		Example("0x76c0")
	})
	Attribute("gasPrice", String, func() {
		Description("The price used to determine the total gas cost. If not provided, an acceptable value will be provided for you.")
		Example("0x9184e72a000")
	})
	Attribute("nonce", String, func() {
		Description("The transaction sequence number from the sending account. If not provided, this will be calculated for you.")
		Example("0x76c0")
	})
	Attribute("to", String, func() {
		Description("The address or smart contract to which the transaction should be sent")
		Example("0xd46e8dd67c5d32be8058bb8eb970870f072445675")
	})
	Attribute("value", String, func() {
		Description("The amount to send. In the units of the smallest denomination for the coin/token (e.g. wei for ethereum)")
		Example("0x9184e72a")
	})

	Required("to", "value")
})
```

## Specifications
```
$ goagen version
goagen v1.3.1
The goa generation tool.

$ lsb_release -r
Release:        16.04
```

Note: I'm using Windows Subsystem for Linux (WSL) with Ubuntu 16.04 and Windows 10 version 1709 build 16299.248

--- a/design/definitions.go
+++ b/design/definitions.go
@@ -616,15 +616,35 @@ func (a *APIDefinition) MediaTypeWithIdentifier(id string) *MediaTypeDefinition 
 // Iteration stops if an iterator returns an error and in this case IterateResources returns that
 // error.
 func (a *APIDefinition) IterateResources(it ResourceIterator) error {
-	names := make([]string, len(a.Resources))
+	res := make([]*ResourceDefinition, len(a.Resources))
 	i := 0
-	for n := range a.Resources {
-		names[i] = n
+	for _, r := range a.Resources {
+		res[i] = r
 		i++
 	}
-	sort.Strings(names)
-	for _, n := range names {
-		if err := it(a.Resources[n]); err != nil {
+	// Iterate parent resources first so that action parameters are
+	// finalized prior to child actions needing them.
+	isParent := func(p, c *ResourceDefinition) bool {
+		par := c.Parent()
+		for par != nil {
+			if par == p {
+				return true
+			}
+			par = par.Parent()
+		}
+		return false
+	}
+	sort.Slice(res, func(i, j int) bool {
+		if isParent(res[i], res[j]) {
+			return true
+		}
+		if isParent(res[j], res[i]) {
+			return false
+		}
+		return res[i].Name < res[j].Name
+	})
+	for _, r := range res {
+		if err := it(r); err != nil {
 			return err
 		}
 	}

